src/System.Slices/System/Buffers/PinnedMemoryHandle.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
```

Comment:
I think it would be better to move this type to System.Buffers namespace. In fact, I think everything, we will move everything related to Memory<T> to this sub namespace. Possibly the only types that will stay in System are Span and ReadOnlySpan.